lib/Resque.php

```diff
@@ -337,4 +337,38 @@ public static function generateJobId()
         return md5(uniqid('', true));
     }
 
+    public static function getInProgressJobsCount() {
+        return self::redis()->hlen('current_jobs');
+    }
+
+    /**
+     * Clean workers that were terminated after timeout
+     *
+     * @return array of unfinished jobs
+     */
+    public static function cleanWorkers() {
+        $notFinishedJobs = [];
+        $workers = self::redis()->sMembers('workers');
+        foreach ($workers as $workerId) {
+            if (strpos($workerId, '-scheduler-') === false && !self::redis()->get('worker:' . $workerId . ':ping')) {
```

Comment:
extract to isSchedulerWorker() and isWorkerAliveByPing()

Reply:
extracted condition into two functions